pkg/operator/resources/batchapi/job.go

```diff
@@ -153,38 +140,47 @@ func uploadJobSpec(jobSpec *spec.Job) error {
 }
 
 func deployJob(apiSpec *spec.API, jobSpec *spec.Job, submission *schema.JobSubmission) {
+	logger, err := operator.GetJobLoggerFromSpec(apiSpec, jobSpec.JobKey)
+	if err != nil {
+		telemetry.Error(err)
+		logging.Logger.Error(err)
+		return
+	}
+
 	totalBatches, err := enqueue(jobSpec, submission)
 	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to enqueue all batches").Error())
+
 		err := errors.FirstError(
-			writeToJobLogStream(jobSpec.JobKey, errors.Wrap(err, "failed to enqueue all batches").Error()),
 			setEnqueueFailedStatus(jobSpec.JobKey),
 			deleteJobRuntimeResources(jobSpec.JobKey),
 		)
 		if err != nil {
 			telemetry.Error(err)
-			errors.PrintError(err)
+			logging.Logger.Error(err)
 		}
 		return
 	}
 
 	if totalBatches == 0 {
 		var errs []error
-		writeToJobLogStream(jobSpec.JobKey, ErrorNoDataFoundInJobSubmission().Error())
+		logger.Error(ErrorNoDataFoundInJobSubmission())
 		if submission.DelimitedFiles != nil {
-			errs = append(errs, writeToJobLogStream(jobSpec.JobKey, "please verify that the files are not empty (the files being read can be retrieved by providing `dryRun=true` query param with your job submission"))
+			logger.Error("please verify that the files are not empty (the files being read can be retrieved by providing `dryRun=true` query param with your job submission")
 		}
 		errs = append(errs, setEnqueueFailedStatus(jobSpec.JobKey))
 		errs = append(errs, deleteJobRuntimeResources(jobSpec.JobKey))
 
 		err := errors.FirstError(errs...)
 		if err != nil {
 			telemetry.Error(err)
-			errors.PrintError(err)
+			logging.Logger.Error(err)
```

Comment:
shouldn't this be `logger.Error`? I'm a bit confused between the two different loggers purpose